### PR TITLE
Migrating to using LightningCSS

### PR DESCRIPTION
**Pull request by Woedenaz**
LightningCSS is just faster and replaces a number of PostCSS plugins, making the number of required package installations lower. The resulting minified CSS file is also, theoretically, slightly faster to load compared to the old one. Haven't exactly tested this myself but it *should* be.

I also made some small changes to a few make files just to make the build process a little more efficient since some commands were being run even when they didn't need to. I'm not exactly the makefile expert so feel free to suggest any better methods if they exist!

### CODE REVIEW

**Comment by Woedenaz** [comment]
Hmm, genuinely not sure why the build is failing. It says `error: Script not found "svgo"` but `bun svgo` has worked just fine in the past. So I'm a little confused there.

File: build/css.mk [hunk 1]
```diff
@@ -12,12 +12,14 @@ dist/css/parts/%: src/css/parts/%
...
 # CSS rules
-dist/css/min/black-highlighter.min.css: src/css/black-highlighter.css $(BUILD_SOURCES) $(CSS_SOURCES) node_modules
+dist/css/min/black-highlighter.min.css: src/css/black-highlighter.css $(CSS_SOURCES) node_modules
 		build/install.sh 644 $< dist/css/black-highlighter-imports.css
 		bun postcss dist/css/black-highlighter-imports.css --config build --env development -o dist/css/black-highlighter.css
 		bun postcss dist/css/black-highlighter-imports.css --config build --env production -o $@
+		touch $@
 
-dist/css/min/normalize.min.css: src/css/normalize.css $(BUILD_SOURCES) node_modules
+dist/css/min/normalize.min.css: src/css/normalize.css node_modules
```
Comment (emmiegit) [inline build/css.mk:21]: Why were these removed? If a build process is updated then those build rules need to be rerun, because it might be different now, right?

File: build/css.mk [hunk 2]
```diff
@@ -12,12 +12,14 @@ dist/css/parts/%: src/css/parts/%
...
 		build/install.sh 644 $< dist/css/normalize.css
 		bun postcss dist/css/normalize.css --config build --env development -r
 		bun postcss dist/css/normalize.css --config build --env production -o $@
+		touch $@
```
Comment (emmiegit) [inline build/css.mk:25]: Shouldn't this be unnecessary, since it writes to the output file each time? Or does `bun` not do anything if it's already up-to-date, leaving the timestamp as-is?

Comment (Woedenaz) [reply build/css.mk:25]: It seems bun does not do anything if it is already up-to-date. Without the `touch $@` those commands seem to get run every time.

File: build/css.mk (hunk 1 above)
Comment (Woedenaz) [reply build/css.mk]: Meant to leave a comment. I wasn't sure why these were included but that makes sense! I've added them back in.

**Review by emmiegit — APPROVED** [review]
